File: src/CandidatosCore.Core/Notificaciones/NotificadorCambios.cs
namespace CandidatosCore.Core.Notificaciones;

using CandidatosCore.Core.Historial;

public sealed class NotificadorCambios
{
    private readonly List<ObservadorCambio> observadores = new();

    public void Suscribir(ObservadorCambio observador) => observadores.Add(observador);

    public void Notificar(RegistroCambio cambio)
    {
        foreach (var observador in observadores)
            observador.Actualizar(cambio);
    }
}

File: src-ts/CandidatosCore.Core/Notificaciones/NotificadorCambios.ts
import type { RegistroCambio } from "../Historial/RegistroCambio";
import type { ObservadorCambio } from "./ObservadorCambio";

export class NotificadorCambios {
    private readonly observadores: ObservadorCambio[] = [];

    public suscribir(observador: ObservadorCambio): void {
        this.observadores.push(observador);
    }

    public notificar(cambio: RegistroCambio): void {
        for (const observador of this.observadores) {
            observador.actualizar(cambio);
        }
    }
}
